### PR TITLE
Add Spanish conjugation source and test file

### PR DESCRIPTION
**Pull request by trifrog**
First attempt to build solution for Spanish conjugation.  This version
only shows present tense for 'yo' as the test case.  Folllow on
submissions will contain other considerations:

(1) Singular / Plural
(2) Masculine / Feminine
(3) Past / Future Tenses
(4) Familiar / Formal

### CODE REVIEW

File: src/spanish.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class Spanish {
+  conjugate(word,info){
+
```
Comment (cecilebertin) [inline src/spanish.js]: @trifrog Nice work!
Could you also add a switch with cases for the tenses? You can just add one case Present tense for now.

**Comment by cecilebertin** [comment]
@trifrog Could you also fix the lint errors?

File: src/spanish.js [hunk 2]
```diff
@@ -1,14 +1,98 @@
...
+};
+const allInfo = {
+  tense: ['present', 'future', 'past'],
+  pronoun: ['yo', 'tú', 'él/ella/Ud.', 'nosotros/nosotras', 'vosotros/vosotras', 'ellos/ellas/Uds.']
+};
+
 class Spanish {
   conjugate (word, info) {
-    const NOUN_ERROR = 'Please input yo, tû, Ud., él, or ella for the noun';
+    const ending = word.slice(word.length - 2, word.length);
```
Comment (songz) [inline src/spanish.js]: `word.slice(-2)`

File: src/spanish.js [hunk 3]
```diff
@@ -1,14 +1,98 @@
...
+    const stem = word.slice(0, -2);
+    const ERROR_MESSAGE = {
+      FOR_TENSE: 'Please select a valid tense: present, future, past.',
```
Comment (songz) [inline src/spanish.js]: keys for object should be consistent. `type` is a good key, `for_tense` is not.
```
const ENDING_ERROR = {
  type: 'ending',
  message: '.....'
};
const ....
```

File: src/spanish.js [hunk 4]
```diff
@@ -1,14 +1,98 @@
...
-      case 'yo':
-        return `${word.slice(0, -2)}o`;
-      default:
-        throw new Error(NOUN_ERROR);
+    if (ending === 'ar' || ending === 'er' || ending === 'ir') {
+      switch (info.tense) {
+        case 'present':
+          return stem.concat(presentTenseConjugationMap[ending][info.pronoun]);
+        case 'future':
+          return word.concat(futureTenseConjugationMap[info.pronoun]);
```
Comment (songz) [inline src/spanish.js]: y is this word?

Comment (trifrog) [reply src/spanish.js]: For future conjugated verbs, the whole word is used rather than the stem.

File: src/spanish.js [hunk 5]
```diff
@@ -1,14 +1,97 @@
...
+const allInfo = {
+  tense: ['present', 'future', 'past'],
+  pronoun: ['yo', 'tú', 'él/ella/Ud.', 'nosotros/nosotras', 'vosotros/vosotras', 'ellos/ellas/Uds.']
+};
+
 class Spanish {
   conjugate (word, info) {
-    const NOUN_ERROR = 'Please input yo, tû, Ud., él, or ella for the noun';
+    const ending = word.slice(-2);
+    const stem = word.slice(0, -2);
```
Comment (songz) [inline src/spanish.js]:
```
let ending = ...
let stem = ....
if (info.tense === 'future') {
  stem = word;
}
return `${stem}${ending}`;
```

File: test/spanish_tests.js [hunk 6]
```diff
@@ -1,18 +1,31 @@
...
+    it('should conjugate present tense correctly', () => {
+      conjugatedWord = spanish.conjugate('hablar', { tense: 'present', pronoun: 'yo' });
       expect(conjugatedWord).to.equal('hablo');
-      
+    });
+  });
+
+  describe('Future Tense', () => {
+    it('should conjugate future tense', () => {
+      conjugatedWord = spanish.conjugate('hablar', { tense: 'future', pronoun: 'él/ella/Ud.' });
```
Comment (songz) [inline test/spanish_tests.js]: more tests! future tense should include at least `ar`, `er`, `ir` endings!

**Pull request merged**